pkg/workload/workload.go

```diff
@@ -893,6 +904,34 @@ func HasConditionWithTypeAndReason(w *kueue.Workload, cond *metav1.Condition) bo
 	return false
 }
 
+func HasFailedNodeAnnotation(w *kueue.Workload) bool {
```

Comment:
Maybe `HasNodeToReplace"? I think the "Annotation" is correct, but it is leaking the internal detail of using the annotation. So that we don't need to change it when we move to the proper workload.status API. wdyt?

Reply:
I agree, done